app/models/dataset/Dataset.scala

```diff
@@ -733,6 +762,53 @@ class DatasetMagsDAO @Inject()(sqlClient: SqlClient)(implicit ec: ExecutionConte
     replaceSequentiallyAsTransaction(clearQuery, insertQueries)
   }
 
+  def updateMagPathsForDataset(datasetId: ObjectId, magPaths: List[MagPathInfo]): Fox[Unit] =
+    for {
+      _ <- Fox.successful(())
+      updateQueries = magPaths.map(pathInfo => {
+        val magLiteral = s"(${pathInfo.mag.x}, ${pathInfo.mag.y}, ${pathInfo.mag.z})"
+        q"""UPDATE webknossos.dataset_mags
+                 SET path = ${pathInfo.path}, realPath = ${pathInfo.realPath}, haslocaldata = ${pathInfo.hasLocalData}
+                 WHERE _dataset = $datasetId
+                  AND datalayername = ${pathInfo.layerName}
+                  AND mag = CAST($magLiteral AS webknossos.vector3)""".asUpdate
+      })
+      composedQuery = DBIO.sequence(updateQueries)
+      _ <- run(
+        composedQuery.transactionally.withTransactionIsolation(Serializable),
+        retryCount = 50,
+        retryIfErrorContains = List(transactionSerializationError)
+      )
+    } yield ()
+
+  def findPathsForDatasetAndDatalayer(datasetId: ObjectId, dataLayerName: String): Fox[List[DatasetMagInfo]] =
+    for {
+      rows <- run(q"""SELECT $columns
+            FROM webknossos.dataset_mags
+            WHERE _dataset = $datasetId
+            AND datalayername = $dataLayerName""".as[DatasetMagsRow])
+      magInfos <- Fox.combined(rows.toList.map(parse))
+      datasetMagInfos = magInfos.map(magInfo =>
+        DatasetMagInfo(datasetId, magInfo.layerName, magInfo.mag, magInfo.path, magInfo.realPath, magInfo.hasLocalData))
+    } yield datasetMagInfos
+
+  def findAllByRealPath(realPath: String): Fox[List[DatasetMagInfo]] =
+    for {
+      rows <- run(q"""SELECT $columns
+            FROM webknossos.dataset_mags
+            WHERE realPath = $realPath""".as[DatasetMagsRow])
+      mags <- Fox.serialCombined(rows.toList)(r => parseMag(r.mag))
+      magInfos = rows.toList.zip(mags).map {
+        case (row, mag) =>
+          DatasetMagInfo(ObjectId(row._Dataset),
+                         row.datalayername,
+                         mag,
+                         row.path.getOrElse("uninitialized"),
+                         row.realpath.getOrElse("uninitialized"),
```

Comment:
Should we use None for this case? I feel like having a custom string literal in the json could cause headaches down the line. What do you think?

Reply:
I did this to avoid having options in the DatasetMagInfo, since this is only a theoretical case either way. The path and realPath are set when the dataset is reported, so it would only be empty between the two report functions that are separate at the moment.

Reply:
I wouldn’t call it a theoretical case, since there is a real time difference between the two report functions. That means that this value will, at least for short durations, occur in the database and downstream functions will have to be able to handle it. If a function lists all datasets that use the same layer, anad does string comparison, this would return all that have "uninitialized", which could become a nonsensical list. I’d rather have the Option if we cannot prevent that this value is ever inserted.